### PR TITLE
Base learning rate decay on training loss

### PR DESCRIPTION
**Pull request by peastman**
This changes it to base learning rate decay on training loss rather than validation loss.  That gives a much cleaner signal for whether it is still learning.  The practical effect is that you can use a smaller value for `lr_patience`, which leads to faster training.

In general, training loss tells you whether it is learning, and the difference between training loss and validation loss tells you whether it is overfitting.  If the training loss stops decreasing, that means you need to reduce the learning rate.  If the training loss is still decreasing but the validation loss stops going down, that means it is overfitting and you should stop.  Reducing the learning rate won't help.

### CODE REVIEW

**Comment by peastman** [comment]
I want to make another push for this change.  Can we at least make it a supported option?  It makes a huge difference to training speed.

I consistently find that the fastest way to train models is to start with a high learning rate, then reduce it aggressively when training stalls (`lr_patience` set to 1 or 2 and `lr_factor` around 0.5).  But that only works when basing it on training loss.  Validation loss is much too noisy, so you have to wait much longer before reducing the learning rate.  That leads to much slower training

**Comment by giadefa** [comment]
I am fine with that, if we can make it an option.

**Comment by peastman** [comment]
This is superseded by #89.
